src/cmap/auth/mongo_credentials.ts

```diff
@@ -89,6 +89,11 @@ export class MongoCredentials {
       }
     }
 
+    if ('gssapiCanonicalizeHostName' in this.mechanismProperties) {
+      this.mechanismProperties.CANONICALIZE_HOST_NAME =
```

Comment:
Let's add an `emitWarningOnce` here to maximize awareness

Reply:
Added.